host-spec/c06-consensus.tm

```diff
@@ -302,28 +369,47 @@
   respectively.
 
   <subsection|Slot Number Calculation>
 
-  It is essential for a block producer to calculate and validate the slot
-  number at a certain point in time. Slots are dividing the time continuum in
-  an overlapping interval. At a given time, the block producer should be able
-  to determine the set of slots which can be associated to a valid block
-  generated at that time. We formalize the notion of validity in the
-  following definitions:
+  It is imperative for the security of the network that each block producer
+  is able to correctly determine the set of slots which can be associated to
+  a valid block generated at a given time without relying on any external
+  clock source (e.g. through the <with|font-shape|italic|Network Time
+  Protocol> or the <with|font-shape|italic|Global Positioning System>). To
+  stay in synchronization each producer is therefore required to periodically
+  estimate its local clock offset in relation to the rest of the network.\ 
+
+  This estimation depends on the two fixed parameters
+  <with|font-series|bold|<math|k>> (def. <reference|defn-prunned-best>) and
+  <math|<with|font-series|bold|s<rsub|cq>>> (def.
+  <reference|defn-chain-quality>). These are choosen based on the results of
+  simulation parameterized to meet certain security bounds, currently
+  assuming a <math|1 s \ >clock drift per day and targeting a probability
+  lower than <math|0.5%> for an adversary to break BABE in 3 years with a
+  resistance against network delay up to <math|<frac*|1|3>> of the slot time.
 
   <\definition>
-    <label|slot-time-cal-tail>The <strong|slot tail>, formally referred to by
-    <math|SlTl> represents the number of on-chain blocks that are used to
-    estimate the slot time of a given slot. This number is set to be 1200.
+    <label|defn-prunned-best>The <with|font-series|bold|prunned best chain>
+    <math|C<rsup|\<#250C\>k>> is the best chain selected according to the
+    best chain selection rule with the last k Blocks prunned. Based on
+    simulation results, we choose <math|k=140>. The
+    <with|font-series|bold|last (probabilistically) finalized block>
+    describes the last block in this prunned best chain.
   </definition>
 
-  Algorithm <reference|algo-slot-time> determines the slot time for a future
-  slot based on the <em|block arrival time> associated with blocks in the
-  slot tail defined in Definition <reference|defn-block-time>.
+  <\definition>
+    <label|defn-chain-quality>The <with|font-series|bold|chain quality>
+    <math|s<rsub|cq>> represents the number of slots hat are used to estimate
+    the local clock offset. Based on simulation results, we choose
+    <math|s<rsub|cq>=3000>.
+  </definition>
+
+  The prerequisite for such a calculation is that each producer stores the
+  arrival time of each block (def. <reference|defn-block-time>).\ 
 
   <\definition>
     <label|defn-block-time>The <strong|block arrival time> of block <math|B>
```

Comment:
Maybe, we should emphasize that the local timer is the computer clock which is not adjusted by any protocol such as NTP

Reply:
Added "... measured by a clock that is otherwise not adjusted by any external protocol."